blackjack/blackjack.go

```diff
@@ -0,0 +1,236 @@
+package blackjack
+
+import (
+	"cards"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+)
+
+type Action int
+
+const (
+	None Action = iota
+	Stand
+	Hit
+)
+
+type Outcome int
+
+const (
+	PlayerBlackjack Outcome = iota
+	PlayerWin
+	PlayerLose
+	PlayerTie
+	PlayerBust
+)
+
+type Game struct {
+	Player Player
+	Dealer Player
+	Shoe   cards.Deck
+}
+
+type Option func(*Game) error
+
+func WithCustomDeck(deck cards.Deck) Option {
+	return func(g *Game) error {
+		g.Shoe = deck
+		return nil
+	}
+}
+
+func NewBlackjackGame(opts ...Option) (*Game, error) {
+	deck := cards.NewDeck(
+		cards.WithNumberOfDecks(1),
+	)
+	random := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	game := &Game{
+		Shoe: deck.Shuffle(random),
+	}
+
+	for _, o := range opts {
+		o(game)
+	}
+
+	fmt.Println(game.Shoe.Cards)
+	return game, nil
+}
+
+func (g *Game) Deal() cards.Card {
```

Comment:
Does it make sense for `Deal` to become a method on `Hand`? Then you'd write code like:

```go
player.Hand.Deal()
dealer.Hand.Deal()
```

Reply:
Is this what you are thinking?


type Hand struct {
	Cards []cards.Card
}

func (h *Hand) Deal(shoe *cards.Deck) {
	var card cards.Card

	card, shoe.Cards = shoe.Cards[0], shoe.Cards[1:]
	shoe.Cards = append(shoe.Cards, card)

	h.Cards = append(h.Cards, card)
}

```
player.Hand.Deal(&g.Shoe)
```